examples/amplify-react/src/master.js

```diff
@@ -0,0 +1,240 @@
+    this.setState({
+      remoteViewSrcObject: srcObject
+    })
+  }
+
+  async start(credentials, params) {
+
+    const {
+      region,
+      endpoint,
+      channelName,
+      clientId,
+      sendAudio,
+      sendVideo,
+      openDataChannel,
+      widescreen,
+      forceTURN,
+      natTraversalDisabled,
+      useTrickleICE,
+    } = params
+
+
+    this.webRtcClient = new WebRtcClient(
+      Role.MASTER,
+      credentials,
+      region,
+      endpoint,
+      channelName,
+      clientId,
+      natTraversalDisabled,
+      forceTURN,
+    )
+
+    const channelARN = await this.webRtcClient.getChannelARN()
+    console.log('[MASTER] Channel ARN: ', channelARN)
+
+    const endpointsByProtocol = await this.webRtcClient.getEndpointsByProtocol()
+    console.log('[MASTER] Endpoints: ', endpointsByProtocol)
+
+    const iceServers = await this.webRtcClient.getIceServers()
+    console.log('[MASTER] ICE servers: ', iceServers)
+
+    this.signalingClient = await this.webRtcClient.getSignalingClient()
+
+    const configuration = {
+      iceServers,
+      iceTransportPolicy: forceTURN ? 'relay' : 'all',
+    };
+
+    const resolution = widescreen ? { width: { ideal: 1280 }, height: { ideal: 720 } } : { width: { ideal: 640 }, height: { ideal: 480 } };
+    const constraints = {
+      video: sendVideo ? resolution : false,
+      audio: sendAudio,
+    };
+
+    // Get a stream from the webcam and display it in the local view.
+    // If no video/audio needed, no need to request for the sources.
+    // Otherwise, the browser will throw an error saying that either video or audio has to be enabled.
+    if (sendVideo || sendAudio) {
+        try {
+            this.localStream = await navigator.mediaDevices.getUserMedia(constraints);
+            this.setLocalViewSrcObject(this.localStream)
+        } catch (e) {
+            console.error('[MASTER] Could not find webcam');
+        }
+    }
+
+    this.signalingClient.on('open', async () => {
+        console.log('[MASTER] Connected to signaling service');
+    });
+
+    this.signalingClient.on('sdpOffer', async (offer, remoteClientId) => {
+        console.log('[MASTER] Received SDP offer from client: ' + remoteClientId);
+
+        // Create a new peer connection using the offer from the given client
+        const peerConnection = new RTCPeerConnection(configuration);
+        this.peerConnectionByClientId[remoteClientId] = peerConnection;
+
+        if (openDataChannel) {
+            this.dataChannelByClientId[remoteClientId] = peerConnection.createDataChannel('kvsDataChannel');
+            peerConnection.ondatachannel = event => {
+                event.channel.onmessage = this.onRemoteDataMessage;
+            };
+        }
+
+        // Poll for connection stats
+        if (!this.peerConnectionStatsInterval) {
+            this.peerConnectionStatsInterval = setInterval(() => peerConnection.getStats().then(this.onStatsReport), 1000);
```

Comment:
Wondering if 1 second would generate a lot of spew? Can we raise to say 5 seconds?

Reply:
This code was taken from the existing example, here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/blob/master/examples/master.js#L136-L139

Whatever value you suggest be implemented here should be implemented in both places.